#Maestria en "Analítica de Datos"
---

Nombre: Oswaldo Salgado Gómez

Código: 79133277

Fecha: 2025 agosto 28

Descripción: Este libro trabaja ejercicios de Análitica/ETL/Estadística Descriptiva


#1.Cargar librerias

In [1]:
import pandas as pd
import sqlite3
import os
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#2. Carga/Obtención de la información

In [3]:
file_path_csv="/content/drive/MyDrive/3. Semestre 3/Big Data/Talleres/T_2/BDUA/Población_Base_de_Datos_Única_de_Afiliados_BDUA_del_régimen_contributivo_20250828.csv"
try:
  df=pd.read_csv(file_path_csv)
except FileNotFoundError:
  print(f"El archivo {file_path_csv} no fue encontrado")
except Exception as e:
    print(f"Ocurrio un error al leer el archivo {e}")

In [4]:
pd.set_option("display.max.columns", None)
display(df.head())

Género Grupo etario Código de la entidad        Nombre de la entidad  \
0  Masculino      45 a 50               ESSC24       COOSALUD EPS S.A. -CM   
1   Femenino         > 75               EPS037              NUEVA EPS S.A.   
2  Masculino      15 a 19               EPS037              NUEVA EPS S.A.   
3   Femenino      15 a 19               EPS017        EPS FAMISANAR S.A.S.   
4  Masculino      45 a 50               ESSC62  ASMET SALUD EPS S.A.S. -CM   

        Régimen Tipo de afiliado   Estado del afiliado  \
0  Contributivo        COTIZANTE                Activo   
1  Contributivo        COTIZANTE                Activo   
2  Contributivo        COTIZANTE  Protección Laboral C   
3  Contributivo     BENEFICIARIO                Activo   
4  Contributivo        COTIZANTE                Activo   

  Condición del beneficiario         Zona de Afiliación  Departamento  \
0                  NO APLICA  Urbana-Cabecera Municipal     MAGDALENA   
1                  NO APLICA                     Urbana       BOLIVAR   
2                  NO APLICA                     Urbana        ARAUCA   
3                 ESTUDIANTE  Urbana-Cabecera Municipal  CUNDINAMARCA   
4                  NO APLICA          Rural - Dispersal         CAUCA   

  Municipio tps_nvl_ssb_id                       Nivel del Sisbén  \
0   PIVIJAY            NaN                                    NaN   
1   CALAMAR              1                   POBLACIÓN CON SISBEN   
2    ARAUCA              N  VÍCTIMAS DEL CONFLICTO ARMADO INTERNO   
3  MOSQUERA            NaN                                    NaN   
4    SUAREZ              1                   POBLACIÓN CON SISBEN   

  Cantidad de registros  
0                     6  
1                     1  
2                     1  
3                     2  
4                     1

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 656598 entries, 0 to 656597
Data columns (total 14 columns):
 #   Column                      Non-Null Count   Dtype 
---  ------                      --------------   ----- 
 0   Género                      656598 non-null  object
 1   Grupo etario                656598 non-null  object
 2   Código de la entidad        656598 non-null  object
 3   Nombre de la entidad        656598 non-null  object
 4   Régimen                     656598 non-null  object
 5   Tipo de afiliado            656598 non-null  object
 6   Estado del afiliado         656598 non-null  object
 7   Condición del beneficiario  656598 non-null  object
 8   Zona de Afiliación          656598 non-null  object
 9   Departamento                656598 non-null  object
 10  Municipio                   656598 non-null  object
 11  tps_nvl_ssb_id              369349 non-null  object
 12  Nivel del Sisbén            369250 non-null  object
 13  Cantidad de registros       6

In [6]:
df.describe()

Género Grupo etario Código de la entidad Nombre de la entidad  \
count      656598       656598               656598               656598   
unique          2           12                   31                   31   
top     Masculino      19 a 45               EPS037       NUEVA EPS S.A.   
freq       332480       156230               278042               278042   

             Régimen Tipo de afiliado Estado del afiliado  \
count         656598           656598              656598   
unique             1                3                   2   
top     Contributivo        COTIZANTE              Activo   
freq          656598           338182              613592   

       Condición del beneficiario Zona de Afiliación Departamento  \
count                      656598             656598       656598   
unique                          3                  6           33   
top                     NO APLICA             Urbana    ANTIOQUIA   
freq                       629227             298692        83198   

          Municipio tps_nvl_ssb_id      Nivel del Sisbén Cantidad de registros  
count        656598         369349                369250                656598  
unique         1035              4                    19                  3034  
top     BOGOTA D.C.              1  POBLACIÓN CON SISBEN                     1  
freq           6251         143052                242862                285920

#3. ETL

In [7]:
df["Cantidad de registros"]=pd.to_numeric(df["Cantidad de registros"], errors="coerce")

In [8]:
columnas_a_convertir =["Departamento", "Municipio", "Código de la entidad","Nivel del Sisbén"]
for columna in columnas_a_convertir:
    if columna in df.columns:
        df[columna]=df[columna].astype(str)


#4.Estadistica Descriptiva

In [9]:
df["Género"].value_counts()

Género
Masculino    332480
Femenino     324118
Name: count, dtype: int64

In [10]:
df_eps=df.groupby("Nombre de la entidad")["Cantidad de registros"].sum()
#Calcular porcentaje de cada EPS
total_registros=df["Cantidad de registros"].sum()
entidad_porcentaje=(df_eps/total_registros)*100
resultado_df=pd.DataFrame({"Cantidad de registros": df_eps, "Porcentaje": entidad_porcentaje})
resultado_df=resultado_df.sort_values(by="Cantidad de registros", ascending=False)
display(resultado_df.head())
#display(df_eps.head())

Cantidad de registros  \
Nombre de la entidad                                                        
NUEVA EPS S.A.                                                  3191312.0   
SALUD TOTAL ENTIDAD PROMOTORA DE SALUD DEL REGI...              1026566.0   
ENTIDAD PROMOTORA DE SALUD SANITAS S.A.S.                        956870.0   
EPS SURAMERICANA S.A.                                            639941.0   
EPS FAMISANAR S.A.S.                                             619655.0   

                                                    Porcentaje  
Nombre de la entidad                                            
NUEVA EPS S.A.                                       37.240613  
SALUD TOTAL ENTIDAD PROMOTORA DE SALUD DEL REGI...   11.979383  
ENTIDAD PROMOTORA DE SALUD SANITAS S.A.S.            11.166074  
EPS SURAMERICANA S.A.                                 7.467711  
EPS FAMISANAR S.A.S.                                  7.230986

5. Creación de Base de Datos SQLITE y Tablas

In [11]:
db_path="/content/drive/MyDrive/3. Semestre 3/Big Data/Talleres/T_2/BDUA/SQLITE/dbua.db"
db_dir=os.path.dirname(db_path)
if os.path.exists(db_dir):
  print(f"La base de datos {db_path} ya existe")
else:
    os.makedirs(db_dir)
    print(f"La base de datos {db_path} fue creada")

La base de datos /content/drive/MyDrive/3. Semestre 3/Big Data/Talleres/T_2/BDUA/SQLITE/dbua.db ya existe


In [12]:
try:
    conn=sqlite3.connect(db_path)
    cursos=conn.cursor() #objeto que va a interactuar con la bd (dml/ddl)
    print("Conectado")
except sqlite3.Error as e:
    print(f"Error al conectar a la base de datos: {e}")

Conectado


#5.1 Creando Tablas Maestras

In [13]:
cursos.execute('''
CREATE TABLE IF NOT EXISTS Departamento (
    Id_Departamento INTEGER PRIMARY KEY AUTOINCREMENT,
    Nombre_Departamento TEXT UNIQUE NOT NULL
)
''')

cursos.execute('''
CREATE TABLE IF NOT EXISTS sisben (
    Id_Sisben INTEGER PRIMARY KEY AUTOINCREMENT,
    Nombre_Sisben TEXT UNIQUE NOT NULL
)
''')

cursos.execute('''
CREATE TABLE IF NOT EXISTS Eps (
    Id_Eps INTEGER PRIMARY KEY AUTOINCREMENT,
    Nombre_Eps TEXT UNIQUE NOT NULL,
    regimen TEX
)
''')

#5.1 Llenado de Tablas Maestras

In [14]:
def cargar_datos_a_tablas_maestras(df, columna, tabla_destino):
    valores_unicos=df[columna].unique()
    for valor in valores_unicos:
        if(pd.notna(valor)):#ingnora valores NaN
            try:
                cursos.execute(f"INSERT OR IGNORE INTO {tabla_destino} (Nombre_{tabla_destino.replace('_','')}) VALUES (?)", (valor,))
            except sqlite3.Error as e:
                print(f"Error al insertar el tabla {tabla_destino}: {e}")

In [17]:
def revisar_contenido_de_una_tabla(tabla_nombre,whereColumna='',whereValor='', limit=10, order_by_columna='', order_asc=True):
    try:
        conn=sqlite3.connect(db_path)
        cursos=conn.cursor()
        cursos.execute("SELECT COUNT(*) FROM "+tabla_nombre)
        total_registros=cursos.fetchone()[0]
        print(f"Total de registros en la tabla {tabla_nombre}: {total_registros}")
        sql="SELECT * FROM "+tabla_nombre
        if len(whereColumna)>0:
            sql+=" WHERE "+whereColumna+"=?"
        if (order_by_columna):
            sql += " ORDER BY " + order_by_columna + (" ASC" if order_asc else " DESC")
        if limit > 0:
            sql += " LIMIT " + str(limit)

        if len(whereColumna) > 0:
            cursos.execute(sql, (whereValor,))
        else:
            cursos.execute(sql)

        resultados = cursos.fetchall()
        for fila in resultados:
            print(fila)

    except sqlite3.Error as e:
        print(f"Error al revisar la tabla {tabla_nombre}: {e}")
    finally:
        if conn:
            conn.close()

#5.3 Tabla "Hija" Municipio

In [18]:
cursos.execute('''
CREATE TABLE IF NOT EXISTS Municipio (
    Id_Municipio INTEGER PRIMARY KEY AUTOINCREMENT,
    Nombre_Municipio TEXT UNIQUE NOT NULL,
    Id_Departamento INTEGER,
    FOREIGN KEY (Id_Departamento) REFERENCES Departamento (Id_Departamento)
    UNIQUE (Nombre_Municipio, Id_Departamento)
)
''')

In [19]:
# Para cada municipio buscar su departamento
def cargar_municipio(df, db_path):
    tot_creados = 0
    conn = None  # Initialize conn to None
    try:
        conn = sqlite3.connect(db_path)
        cursos = conn.cursor()
        for index, row in df.iterrows():
            departamento = row["Departamento"]
            municipio = row["Municipio"]
            cursos.execute("SELECT Id_Departamento FROM Departamento WHERE Nombre_Departamento=?", (departamento,))
            departamento_id = cursos.fetchone()
            if departamento_id:
                departamento_id = departamento_id[0]
                try:
                    cursos.execute("INSERT OR IGNORE INTO Municipio (Nombre_Municipio, Id_Departamento) VALUES (?,?)", (municipio, departamento_id))
                    tot_creados += 1
                except sqlite3.IntegrityError:
                    pass  # Ignore duplicate entries
            else:
                print(f"ERROR: Departamento {departamento} para el municipio {municipio} no encontrado")
        conn.commit()
    except sqlite3.Error as e:
        print(f"Database error: {e}")
    finally:
        if conn:
            conn.close()
    return tot_creados

#5.3 Carga EPS

In [20]:
def cargar_eps(df,db_path):
    conn=sqlite3.connect(db_path)
    cursos=conn.cursor()
    total_creados=0
    valores_unicos=df["Código de la entidad, 'Nombre de la entidad", 'Regimen'].drop_duplicate()
    for index, row in valores_unicos.iterrows():
        codigo_eps=row["Código de la entidad"]
        nombre_eps=row["Nombre de la entidad"]
        try:
            cursos.execute("INSERT OR IGNORE INTO Eps (Nombre_Eps, Id_Eps) VALUES (?,?,?)", (nombre_eps, codigo_eps, regimen))
            total_creados+=1
            print(f"Creando {total_creados} => {nombre_eps} con el código {codigo_eps}")
        except sqlite3.sqlite3.IntegrityError:
            print(f"No fue posible crear la EPS {nombre_eps} con el codigo {codigo_eps} due to {e}")
            pass
    conn.commit()
    conn.close()
    return total_creados
conn.commit()
conn.close()

#5.5 Crear tabla nieta "Detalle"

0.   Género                      656598 non-null  object
 1.   Grupo etario                656598 non-null  object
 2.   Código de la entidad        656598 non-null  object
 3.   Nombre de la entidad        656598 non-null  object
 4.   Régimen                     656598 non-null  object
 5.   Tipo de afiliado            656598 non-null  object
 6.   Estado del afiliado         656598 non-null  object
 7.   Condición del beneficiario  656598 non-null  object
 8.   Zona de Afiliación          656598 non-null  object
 9.   Departamento                656598 non-null  object
 10.  Municipio                   656598 non-null  object
 11.  tps_nvl_ssb_id              369349 non-null  object
 12.  Nivel del Sisbén            369250 non-null  object
 13.  Cantidad de registros       656598 non-null  object

In [21]:
conn=sqlite3.connect(db_path)
cursos=conn.cursor()
cursos.execute('''
CREATE TABLE IF NOT EXISTS Municipio (
    Id_Detalle INTEGER PRIMARY KEY AUTOINCREMENT,
    Id_Municipio INTEGER,
    Id_Eps INTEGER,
    Id_Sisben INTEGER,
    Genero Text,
    Grupo_etario TEXT,
    Codigo_entidad INTEGER,
    Nombre_entidad TEXT,
    Regimen TEXT,
    Tipo_afiliado TEXT,
    Estado_afiliado TEXT,
    Condicion_beneficiario TEXT,
    Zona_afiliacion TEXT,
    FOREIGN KEY (Id_Municipio) REFERENCES Municipio (Id_Municipio),
    FOREIGN KEY (Id_Eps) REFERENCES Eps (Id_Eps),
    FOREIGN KEY (Id_Sisben) REFERENCES Sisben (Id_Sisben)
)
''')
conn.commit()
conn.close()

In [23]:
def cargar_detalle(df,db_path):
    conn=sqlite3.connect(db_path)
    cursos=conn.cursor()
    total_creados=0
    for index, row in df.iterrows():
        municipio =row["Municipio"]
        departamento=row["Departamento"]
        eps=row["Código de la entidad"]
        if (row['Nivel del Sisbén'] is None):
            id_sisben=0
        else:
            sisben=row["Nivel del Sisbén"]
            cursos.execute("SELECT Id_Sisben FROM Sisben WHERE Nombre_Sisben=?", (sisben,))
            sisben_id=cursos.fetchone()
            if sisben_id:
                id_sisben=sisben_id[0]
            else:
                print(f"ERROR: Sisben {sisben} para el municipio {municipio} no encontrado")
# Para obtener el id_ del municipio
        cursos.execute("SELECT M.Id_Municipio FROM Municipio M JOIN Departamento D ON M.Id_Departamento = D.Id_Departamento WHERE M.Nombre_Municipio=? AND D.Nombre_Departamento=?",(municipio,departamento))
        municipio_id=cursos.fetchone()
        if municipio_id:
            municipio_id=municipio_id[0]
        else:
            print(f"ERROR: Municipio {municipio} para el departamento {departamento} no encontrado")

    # Guarda cada registro de la tabla

6. Integrar con Tutelas Radicadas

In [24]:
from concurrent.futures import ThreadPoolExecutor, as_completed
url_folder_csv="/content/drive/MyDrive/3. Semestre 3/Big Data/Talleres/T_2"
columnas_nombre=["clasificacion",
                 "num_expediente",
                 "fecha_radica",
                 "demandante",
                 "demandado",
                 "primera_instancia",
                 "segunda_instancia",
                 "num_23Digitos",
                 "providencia",
                 "fecha_setencia"]
df_tutelas=pd.DataFrame(columns=columnas_nombre)
display(df_tutelas)

Empty DataFrame
Columns: [clasificacion, num_expediente, fecha_radica, demandante, demandado, primera_instancia, segunda_instancia, num_23Digitos, providencia, fecha_setencia]
Index: []

In [25]:
def cargar_data_desde_archivo_csv(ruta_archivo, archivo_nombre):
    try:
        df_temporal=pd.read_csv(ruta_archivo,sep=";",header=None, encoding="latin-1", on_bad_lines="skip")
        #Agregar
        df_temporal.columns=columnas_nombre
        if (len(df_temporal.columns)==len(columnas_nombre)):
            print(f" archivo{archivo_nombre} carga exitosamente")
            return df_temporal
        else:
            print(f"archivo no trabajado {archivo_nombre} no tiene 10 columnas")
            return None

    except Exception as e:
        print(f"Error al procesar el archivo {ruta_archivo}: {str(e)}")
        return None

In [27]:
sd_list=[] #lista de df resultante de los datos de cada uno de los csv
listado_archivos = [f for f in os.listdir(url_folder_csv) if f.endswith('.csv')]
with ThreadPoolExecutor() as executor:
    future_to_file={executor.submit(cargar_data_desde_archivo_csv, os.path.join(url_folder_csv, archivo_nombre), archivo_nombre): archivo_nombre for archivo_nombre in listado_archivos}
    for future in as_completed(future_to_file):
        df=future.result()
        if df is not None:
            sd_list.append(df)

# Concatenate all dataframes in the list
if sd_list:
    df_tutelas_parallel = pd.concat(sd_list, ignore_index=True)
    # Replace "NULL" with None
    df_tutelas_parallel.replace("NULL", None, inplace=True)
    display(df_tutelas_parallel.head())

 archivotutelas_con_providencias_2021.csv carga exitosamente
 archivotutelas_con_providencias_2022.csv carga exitosamente


/tmp/ipython-input-1706949637.py:3: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.
  df_temporal=pd.read_csv(ruta_archivo,sep=";",header=None, encoding="latin-1", on_bad_lines="skip")


 archivotutelas_con_providencias_2025.csv carga exitosamente
 archivotutelas_con_providencias_2023.csv carga exitosamente
 archivotutelas_con_providencias_2024.csv carga exitosamente


clasificacion num_expediente         fecha_radica  \
0  Acciones de Tutela       T8531729  2021-12-16 00:00:00   
1     Acciones de Tutela       T8531726  2021-12-16 00:00:00   
2     Acciones de Tutela       T8531728  2021-12-16 00:00:00   
3     Acciones de Tutela       T8531723  2021-12-16 00:00:00   
4     Acciones de Tutela       T8531727  2021-12-16 00:00:00   

                                          demandante  \
0  VAZQUEZ CUADROS BAYRON EMILIO Y OTROS EN REPRE...   
1                             ESCORCIA JORGE ARMANDO   
2                       ARCINIEGAS MEDRANO CLEMENCIA   
3                        SERRANO ROJAS CAMILO ANDRES   
4                        GOMEZ QUINTERO EDILSA MARIA   

                                      demandado  \
0  SOCIEDAD DE ACTIVOS ESPECIALES S.A.E. S.A.S.   
1             ALCALDIA MUNICIPAL DE SANTA LUCIA   
2                           COLPENCIONES Y OTRO   
3            SECRETARIA DE MOVILIDAD DE CAQUEZA   
4                           CAJACOPI EPS Y OTRO   

                                   primera_instancia  \
0  BOGOTA,CUNDINAMARCA, TRIBUNAL SUPERIOR DE BOGO...   
1  SANTA LUCIA,ATLANTICO, JUZGADO 1 PROMISCUO MUN...   
2  BARRANQUILLA,ATLANTICO, JUZGADO 11 LABORAL DEL...   
3  CAQUEZA,CUNDINAMARCA, JUZGADO 1 PROMISCUO MUNI...   
4        CURUMANI,CESAR, JUZGADO PROMISCUO MUNICIPAL   

                                   segunda_instancia num_23Digitos  \
0                                                 --            --   
1  SABANALARGA,ATLANTICO, JUZGADO 2 PROMISCUO DEL...            --   
2  BARRANQUILLA,ATLANTICO, TRIBUNAL SUPERIOR SALA...            --   
3                                                 --            --   
4       CHIRIGUANA,CESAR, JUZGADO PENAL DEL CIRCUITO            --   

  providencia           fecha_setencia  
0          --                      NaN  
1          --                      NaN  
2    T-274/22  2022-07-28 00:00:00.000  
3          --                      NaN  
4          --                      NaN

In [28]:
df_tutelas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   clasificacion      0 non-null      object
 1   num_expediente     0 non-null      object
 2   fecha_radica       0 non-null      object
 3   demandante         0 non-null      object
 4   demandado          0 non-null      object
 5   primera_instancia  0 non-null      object
 6   segunda_instancia  0 non-null      object
 7   num_23Digitos      0 non-null      object
 8   providencia        0 non-null      object
 9   fecha_setencia     0 non-null      object
dtypes: object(10)
memory usage: 132.0+ bytes


In [30]:
df_tutelas_eps=df_tutelas_parallel[df_tutelas_parallel['demandado'].str.contains('EPS', na=False)]
display(df_tutelas_eps.head())

clasificacion num_expediente         fecha_radica  \
4   Acciones de Tutela       T8531727  2021-12-16 00:00:00   
10  Acciones de Tutela       T8531722  2021-12-16 00:00:00   
17  Acciones de Tutela       T8531705  2021-12-16 00:00:00   
21  Acciones de Tutela       T8531702  2021-12-16 00:00:00   
25  Acciones de Tutela       T8531700  2021-12-16 00:00:00   

                                           demandante             demandado  \
4                         GOMEZ QUINTERO EDILSA MARIA   CAJACOPI EPS Y OTRO   
10  ARAGON CORREA ANDREA AO DE NUÑEZ BARACALDO KE...    CONVIDA EPS T OTRO   
17                      QUIROZ RODRIGUEZ LIDIS ESTHER  LA NUEVA EPS Y OTROS   
21                          HERNANDEZ GOMEZ SEBASTIAN          COOSALUD EPS   
25                                  ROZO PARRA WILMAR     CAPITAL SALUD EPS   

                                    primera_instancia  \
4         CURUMANI,CESAR, JUZGADO PROMISCUO MUNICIPAL   
10  CAQUEZA,CUNDINAMARCA, JUZGADO 1 PROMISCUO MUNI...   
17        PAILITAS,CESAR, JUZGADO PROMISCUO MUNICIPAL   
21  CHIGORODO,ANTIOQUIA, JUZGADO 1 PROMISCUO MUNIC...   
25        ACACIAS,META, JUZGADO 1 PROMISCUO MUNICIPAL   

                               segunda_instancia num_23Digitos providencia  \
4   CHIRIGUANA,CESAR, JUZGADO PENAL DEL CIRCUITO            --          --   
10                                            --            --          --   
17  CHIRIGUANA,CESAR, JUZGADO PENAL DEL CIRCUITO            --          --   
21  CHIRIGUANA,CESAR, JUZGADO PENAL DEL CIRCUITO            --          --   
25                                            --            --          --   

   fecha_setencia  
4             NaN  
10            NaN  
17            NaN  
21            NaN  
25            NaN

## 6.2 Extraer las EPS dbua.db (sqlite)

In [31]:
conn= sqlite3.connect(db_path)
df_eps=pd.read_sql_query("SELECT * FROM Eps", conn)
conn.close()
display(df_eps.head())

AttributeError: module 'pandas' has no attribute 'read'

## Contar las tutelas en contra de cada EPS (df_eps)

In [ ]:
df_eps['total_tutelas']=0
demandados_encontrados= set()
for index, row in df_tutelas_eps.iterrows():
    eps_nombre=row['Nombre_Eps']
    maching_tutelas= df_tutelas_eps[df_tutelas_eps['demandado'], str.contains(eps_nombre, na=False, cas=False)] #Mejorar el código "NUEVA EPS" para que pueda encontrar variadades de manera que se escribe en la base de datos
    total_tutelas=maching_tutelas.shape[0]
    #df_eps.at[index,'total_tutelas']=total_tutelas
    df_eps.loc[df_eps['Nombre_Eps']==eps_nombre, 'total_tutelas']=total_tutelas
    demandados_encontrados.update(maching_tutelas['demandado'].tolist())

demandados_no_encontrados=set(df_tutelas_eps['demandado'].unique())-demandados_encontrados
print(f"EPS demandadas no encontrados:{len[demandados_no_encontrados]}")
print(f"EPS demandados encontrados:{len[demandados_encontrados]}")
for demandado in demandados_no_encontrados:
    print(demandado)
display(df_eps.head())